integrations/transportation/wizzair.js

```diff
@@ -0,0 +1,187 @@
+import moment from 'moment';
+import request from 'superagent';
+import { ACTIVITY_TYPE_TRANSPORTATION, TRANSPORTATION_MODE_PLANE } from '../../definitions';
+import { HTTPError, AuthenticationError, ValidationError } from '../utils/errors';
+
+const API_VERSION_URL = 'https://wizzair.com/static/metadata.json';
+const agent = request.agent();
+const HISTORY_API_FETCH_LIMIT = 5;
+
+async function getApiVersionUrl() {
+  return fetch(API_VERSION_URL).then(response => response.text());
+}
+
+let API_URL;
+let BASE_URL;
+let LOGIN_URL;
+let ITINERARY_URL;
+let BOOKINGS_URL;
+
+async function logIn(username, password) {
+  // urls have to be assigned here as they're asynchronous and have to be awaited
+  await Promise.all(
+    API_URL = await getApiVersionUrl(),
+    BASE_URL = JSON.parse(API_URL.trim()).apiUrl,
+    LOGIN_URL = `${await BASE_URL}/customer/login`,
+    ITINERARY_URL = `${await BASE_URL}/booking/itinerary`,
+    BOOKINGS_URL = `${await BASE_URL}/customer/mybookings`,
+  );
+
+  const res = await agent
+    .post(LOGIN_URL)
+    .type('application/json')
+    .set('Accept', '*/*')
+    .send({
+      username,
+      password,
+      isAgencyLogin: false,
+      captchaResponse: '',
+      languageCode: 'en-gb',
+    });
+
+  if (!res.ok) {
+    throw new AuthenticationError(`Error while logging in. ${res.body}`);
+  }
+  return {};
+}
+
+async function getPastBookings() {
+  const pastBookings = await agent
+    .post(BOOKINGS_URL)
+    .type('application/json')
+    .set('Accept', '*/*')
+    .send({
+      flightdestination: '',
+      flightorigin: '',
+      pnr: '',
+    });
+
+  if (!pastBookings.ok) {
+    const text = await pastBookings.text();
+    throw new HTTPError(text, pastBookings.status);
+  }
+
+  return { 
+    pastBookings: pastBookings.body.pastBookings, 
+    // totalCount: pastBookings.body.length,
+  };
+}
+
+async function getAllFlights(booking) {
+  const allFlights = [];
+
+  await Promise.all(booking.map(async (singleFlight) => {
+    await agent
+      .post(ITINERARY_URL)
+      .type('application/json')
+      .set('Accept', '*/*')
+      .send({
+        keepBooking: false,
+        lastName: singleFlight.contactLastName,
+        pnr: singleFlight.recordLocator,
+      })
+      .then(
+        (res) => {
+          allFlights.push({
+            id: `pnr${res.body.pnr}fn${res.body.outboundFlight.flightNumber}`,
+            flight: res.body.outboundFlight || false, // not every booking has both an outbound and a return flight, hence false
+            pnr: res.body.pnr,
+          }, {
+            id: `pnr${res.body.pnr}fn${res.body.returnFlight.flightNumber}`,
+            flight: res.body.returnFlight || false,
+            pnr: res.body.pnr,
+          });
+        }, ((res) => {
+          throw new HTTPError(`Error while fetching flights. ${res.body}`);
```

Comment:
Can you make this HTTPError use the status and text like the other HTTPError you created?